### PR TITLE
tests(skymp5-server): unit tests for iterating GRUP children in lib_espm

### PR DESCRIPTION
**Pull request by nic11**
See #100 for details

### CODE REVIEW

**Comment by nic11** [comment]
Moved big checklist from this PR to a chacklist issue. Plans may get even bigger!

**Comment by Pospelove** [comment]
Please replace C-style casts with C++ equivalents like `reinterpret_cast`

File: skymp5-server/cpp/lib_espm/espm.h [hunk 1]
```diff
@@ -94,10 +124,17 @@ class GroupHeader
...
-  void ForEachRecord(const RecordVisitor& visitor)
-    const noexcept; // Return true from visitor to break loop
+  using RecordVisitor = std::function<bool(const espm::RecordHeader*)>;
 
+  // Accepts a visitor, which can contain custom code used to recursively
+  // iterate child records. All tree leafs will be visited. Return true from
+  // visitor to break loop.
+  void ForEachRecordRecursive(const RecordVisitor& visitor) const noexcept;
+
+  uint32_t GetGroupLabelAsUint() const noexcept
```
Comment (Pospelove) [inline skymp5-server/cpp/lib_espm/espm.h:134]: Should be moved to .cpp

Comment (nic11) [reply skymp5-server/cpp/lib_espm/espm.h]: done

**Pull request merged**